### PR TITLE
[WEEKLY RELEASE] HotShot 0.5.57

### PR DESCRIPTION
**Pull request by jparr721**
Closes #<ISSUE_NUMBER>

### CODE REVIEW

File: sequencer/src/persistence/fs.rs [hunk 1]
```diff
@@ -408,6 +412,86 @@ impl SequencerPersistence for Persistence {
             },
         )
     }
+    async fn append_quorum_proposal(
+        &mut self,
+        proposal: &Proposal<SeqTypes, QuorumProposal<SeqTypes>>,
+    ) -> anyhow::Result<()> {
+        let view_number = proposal.data.view_number().u64();
```
Comment (jparr721) [inline sequencer/src/persistence/fs.rs:419]: Thoughts @jbearer?

File: sequencer/src/persistence/fs.rs [hunk 2]
```diff
@@ -408,6 +412,86 @@ impl SequencerPersistence for Persistence {
...
+        let dir_path = self.proposals_dir_path();
+
+        fs::create_dir_all(dir_path.clone()).context("failed to create proposals dir")?;
+
+        let file_path = dir_path.join(view_number.to_string()).with_extension("txt");
+        self.replace(
+            &file_path,
+            |_| {
+                // Don't overwrite the file, we want to append since we can get multiple
+                // saved quorum proposals.
```
Comment (jbearer) [inline sequencer/src/persistence/fs.rs:429]: Not for the same view right? I think not overwriting the file is fine, but it's not going to cause us to append, it's just going to keep what was already there

File: sequencer/src/persistence/fs.rs [hunk 3]
```diff
@@ -408,6 +412,86 @@ impl SequencerPersistence for Persistence {
...
+    async fn append_quorum_proposal(
+        &mut self,
+        proposal: &Proposal<SeqTypes, QuorumProposal<SeqTypes>>,
+    ) -> anyhow::Result<()> {
+        let view_number = proposal.data.view_number().u64();
+        let dir_path = self.proposals_dir_path();
+
+        fs::create_dir_all(dir_path.clone()).context("failed to create proposals dir")?;
+
+        let file_path = dir_path.join(view_number.to_string()).with_extension("txt");
```
Comment (jbearer) [inline sequencer/src/persistence/fs.rs:424]: Do we ever prune old ones?

File: sequencer/src/persistence/sql.rs [hunk 4]
```diff
@@ -452,6 +455,30 @@ impl SequencerPersistence for Persistence {
             .transpose()
     }
 
+    async fn load_quorum_proposals(
+        &self,
+    ) -> anyhow::Result<Option<BTreeMap<ViewNumber, Proposal<SeqTypes, QuorumProposal<SeqTypes>>>>>
+    {
+        let rows = self
+            .db
+            .query_static("SELECT * FROM quorum_proposals")
```
Comment (jbearer) [inline sequencer/src/persistence/sql.rs:464]: We need a migration to create this table

**Review by jbearer — CHANGES_REQUESTED** [review]

File: sequencer/src/persistence/fs.rs (hunk 3 above)
Comment (jbearer) [reply sequencer/src/persistence/fs.rs:424]: If it is correct to delete anything equal or less than the decided view (I don't know if that's what we want, but if it is) it should be to just add this to the `delete_files` calls in `collect_garbage`. And likewise the `DELETE` statements in `collect_garbage` in `persistence/sql.rs`

File: sequencer/src/persistence/fs.rs (hunk 2 above)
Comment (jparr721) [reply sequencer/src/persistence/fs.rs:429]: Sorry, this was an old comment. It'll write a new entry per view. I can fix this.

File: sequencer/src/persistence/fs.rs (hunk 3 above)
Comment (jparr721) [reply sequencer/src/persistence/fs.rs:424]: Yeah I think that's the case, I can update that.

File: sequencer/src/persistence/fs.rs (hunk 1 above)
Comment (jbearer) [reply sequencer/src/persistence/fs.rs:420]: Looks reasonable

File: sequencer/src/persistence/sql.rs [hunk 5]
```diff
@@ -452,6 +458,30 @@ impl SequencerPersistence for Persistence {
...
+            .await?;
+
+        Ok(Some(BTreeMap::from_iter(
+            rows.map(|row| {
+                let row = row?;
+                let view: i64 = row.get("view");
+                let view_number: ViewNumber = ViewNumber::new(view.try_into()?);
+                let bytes: Vec<u8> = row.get("proposal");
```
Comment (jbearer) [inline sequencer/src/persistence/sql.rs:475]: Should be `data` now I think

**Review by jbearer — APPROVED** [review]
Looks good

**Pull request merged**